core/src/actors/codec.rs

```diff
@@ -53,11 +56,19 @@ impl Encoder for P2PCodec {
     type Error = io::Error;
 
     /// Method to encode a response into bytes
-    fn encode(&mut self, msg: Response, _dst: &mut BytesMut) -> Result<(), Self::Error> {
+    fn encode(&mut self, msg: Response, dst: &mut BytesMut) -> Result<(), Self::Error> {
         info!("Encoding {:?}", msg);
 
-        // TODO: Encoder to be completed
+        let Response::Message(bytes) = msg;
 
+        let mut encoded_msg = vec![];
+        let header: u16 = bytes.len() as u16;
```

Comment:
What happens when `bytes.len()` doesn't fit in a u16? Is there a max message size?

```rust
if bytes.len() > u16::max_value() as usize {
    return Err(...);
}
```